google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -362,6 +368,13 @@ public ApiFuture<Void> closeAsync() {
         statementExecutor.shutdown();
         leakedException = null;
         spannerPool.removeConnection(options, this);
+        try {
+          if (!databaseAdminClient.isTerminated() || !databaseAdminClient.isShutdown()) {
+            databaseAdminClient.close();
+          }
+        } catch (RuntimeException ex) {
+          throw SpannerExceptionFactory.newSpannerException(ex);
```

Comment:
Are we sure that we want to propagate this exception? Or should we just ignore it? Are there any potential things that the user might have done wrong that could cause an exception here? If yes, then it is probably best to propagate it. If this is more of a 'better-safe-than-sorry' and/or 'this sometimes randomly causes an exception, no idea why'-kind of situation, then I would suggest that we ignore the exception.

But see also above: I don't think we should create a separate `DatabaseAdminClient` for each connection that is created.